Minio/ApiEndpoints/IObjectOperations.cs

```diff
@@ -30,9 +31,8 @@ public interface IObjectOperations
         /// </summary>
         /// <param name="bucketName">Bucket to retrieve object from</param>
         /// <param name="objectName">Name of object to retrieve</param>
-        /// <param name="callback">A stream will be passed to the callback</param>
         /// <param name="cancellationToken">Optional cancellation token to cancel the operation</param>
-        Task GetObjectAsync(string bucketName, string objectName, Action<Stream> callback, CancellationToken cancellationToken = default(CancellationToken));
+        Task<byte[]> GetObjectAsync(string bucketName, string objectName, CancellationToken cancellationToken = default(CancellationToken));
```

Comment:
If we use tasks, then we must abandon callbacks, because then there is no sense in asynchronous methods. The Stream is an abstract class. If we download an object, it's always an array of bytes or memory stream. I will give an example. There is an Akavache library for xamarin, which works with the application's cache. This library has a [method](https://github.com/akavache/Akavache/blob/master/src/Akavache/Portable/RelativeTimeMixin.cs#L20) of downloading data from the Internet.